File: Tyche.Android/bet/src/main/java/com/felipearpa/tyche/bet/application/GetPoolGamblerBetsUseCase.kt
package com.felipearpa.tyche.bet.application

import com.felipearpa.tyche.bet.domain.PoolGamblerBetRepository
import javax.inject.Inject

class GetPoolGamblerBetsUseCase @Inject constructor(private val poolGamblerBetRepository: PoolGamblerBetRepository) {

    suspend fun execute(
        poolId: String,
        gamblerId: String,
        next: String? = null,
        searchText: String? = null
    ) =
        poolGamblerBetRepository.getPoolGamblerBets(
            poolId = poolId,
            gamblerId = gamblerId,
            next = next,
            searchText = searchText
        )
}

File: Tyche.iOS/DataBet/Sources/DataBet/Application/GetPoolGamblerBetsUseCase.swift
import Core

public class GetPoolGamblerBetsUseCase {
    private let poolGamblerBetRepository: PoolGamblerBetRepository
    
    public init(poolGamblerBetRepository: PoolGamblerBetRepository) {
        self.poolGamblerBetRepository = poolGamblerBetRepository
    }
    
    public func execute(
        poolId: String,
        gamblerId: String,
        next: String? = nil,
        searchText: String? = nil
    ) async -> Result<CursorPage<PoolGamblerBet>, Error> {
        return await poolGamblerBetRepository.getPoolGamblerBets(
            poolId: poolId, 
            gamblerId: gamblerId,
            next: next,
            searchText: searchText
        )
    }
}
